src/Uno.UI/UI/Xaml/Controls/Primitives/Selector.cs

```diff
@@ -28,6 +28,11 @@ public partial class Selector : ItemsControl
 		/// </summary>
 		internal virtual bool IsSingleSelection => true;
 
+		/// <summary>
+		/// Templates for which it's known that the template root doesn't qualify as a container.
+		/// </summary>
+		private readonly HashSet<DataTemplate> _itemTemplatesThatArentContainers = new HashSet<DataTemplate>();
```

Comment:
Should we clear this if the `ItemTemplateSelector` changes ?

Reply:
Yes! Fixed